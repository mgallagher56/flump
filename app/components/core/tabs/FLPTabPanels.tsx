import React from 'react';

import type { TabPanelsProps } from '@chakra-ui/react';
import { TabPanels } from '@chakra-ui/react';

const FLPTabPanels: React.FC<TabPanelsProps> = (props) => {
  return <TabPanels {...props}>{props.children}</TabPanels>;
};

export default FLPTabPanels;
