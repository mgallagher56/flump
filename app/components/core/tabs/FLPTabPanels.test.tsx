import React from 'react';

import { Tabs } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import FLPTabPanels from './FLPTabPanels';

const tabWrapper = (children: React.ReactNode): React.ReactElement => <Tabs>{children}</Tabs>;

describe('FLPTabList', () => {
  test('renders correctly', () => {
    const { container } = render(
      tabWrapper(
        <FLPTabPanels>
          <div>Tab panel 1</div>
          <div>Tab panel 2</div>
          <div>Tab panel 3</div>
        </FLPTabPanels>
      )
    );
    expect(container).toMatchSnapshot();
  });
});
