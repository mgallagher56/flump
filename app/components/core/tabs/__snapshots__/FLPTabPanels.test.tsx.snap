// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPTabList > renders correctly 1`] = `
<div>
  <div
    class="chakra-tabs css-79elbk"
  >
    <div
      class="chakra-tabs__tab-panels css-8atqhb"
    >
      <div>
        Tab panel 1
      </div>
      <div>
        Tab panel 2
      </div>
      <div>
        Tab panel 3
      </div>
    </div>
  </div>
</div>
`;
